nixos/modules/services/misc/gitea.nix

```diff
@@ -435,9 +474,12 @@ in
 
     users.groups.gitea = {};
 
-    warnings = optional (cfg.database.password != "")
-      ''config.services.gitea.database.password will be stored as plaintext
-        in the Nix store. Use database.passwordFile instead.'';
+    warnings =
+      optional (cfg.database.password != "") ''
+        config.services.gitea.database.password will be stored as plaintext in the Nix store. Use database.passwordFile instead.'' ++
+      optional (cfg.extraConfig != null) ''
+        services.gitea.`extraConfig` is deprecated, please use services.gitea.`settings`.
```

Comment:
One tiny thing here is that I'd change `config.services.gitea.database.password` to `services.gitea.database.password` for consistency

Reply:
I was planning to remove the `password` option as part of https://github.com/NixOS/nixpkgs/issues/24288 shortly after this PR.

Reply:
+1 to that!